README.md

```diff
@@ -1,35 +1,259 @@
+
+- **directory**: absolute path to a directory that contains a sqitch plan
+- **registry**: (optional) the sqitch registry to use, default is "sqitch"
+- **extra_args**: (optional) extra args for sqitch deploy command, default is "--verify"
+- **engine**: (optional) database engine to use, default is pg
+
+### sqitch_migrations_sql_jobs
+
+An object containing SQL queries that should be executed periodically.
+
+```yml
+sqitch_migrations_sql_jobs:
+  db_admin_user:  # used to identify the db connection to use
+    - file: "/tmp/test.sql"  # absolute path to file containing sql to be run periodically
+      schema: test  # database schema for this to be run in
+      minute: "*5"  # cron config for every 5 minutes
+```
+
+As you can tell, it is an object/dictionary where each entry contains a list of other objects.  The structure of these inner objects is as follows:
+
+- **file**: absolute path to the sql file
+- **schema**: (Optional) the database schema to use, default is "public
+- **month**: (Optional) cron config for month, default is "*"
+- **weekday**: (Optional) cron config for weekday, default is "*"
+- **day**: (Optional) cron config for day, default is "*"
+- **hour**: (Optional) cron config for hour, default is "*"
+- **minute**: (Optional) cron config for minute, default is "*"
+
+## Dependencies
+
+- **onaio.sqitch**: for installing sqitch
+- **adamyala.ansible-role-clone-repo**: for cloning git repositories
+
+You can install these by running the following ansible command:
+
+```sh
+ansible-galaxy install -r requirements.yml
+```
+
+## Example Playbook
 
 ```yml
 - hosts: servers
   roles:
-    - { role: username.rolename, x: 42 }
+    - role: ansible-sqitch-migrations
+      vars:
+        - sqitch_migrations_system_dependencies:
+            - git
+        - sqitch_migrations_repositories:
+            - url: git@github.com:onaio/data-solutions.git
+              destination: "{{ sqitch_migrations_home }}/data-solutions"
+              version: master
+            - url: git@github.com:OpenSRP/opensrp-reveal-datawarehouse.git
+              destination: "{{ sqitch_migrations_home }}/reveal-datawarehouse"
+              version: master
+        - sqitch_migrations_db_connections:
+            db_admin:
+              user: admin
+              password: hunter2
+              host: localhost
+              port: 5432
+              database: productiondb
+            db_user:
+              user: regular
+              password: hunter2
+              host: localhost
+              port: 5432
+              database: productiondb
+        - sqitch_migrations_sql_for_setup:
+            db_admin:
+              - "{{ sqitch_migrations_home }}/data-solutions/OpenSRP/2-reveal/setup/extensions.sql"
+        - sqitch_migrations_sqitch_plans:
+            db_admin:
+              - directory: "{{ sqitch_migrations_home }}/data-solutions/examples/sqitch_test"
+            db_user:
```

Comment:
This one is a little scary to me - if we start assembling a lot of micro-migrations together in playbooks vs data-solutions repo, we won't be able to test stuff very well without a lot of infrastructure.

Do we want to push more of this down to data solutions - you can definitely use subdirectories and symlinks, for example, for shared migration piece.  Alternately copy-paste, if each project is really a separate fork at the time-of-creation, could make sense in this case.

Reply:
I've actually been really torn about how the Reveal migrations are currently structured.  I might change it.  Maybe.  Anyway, I think sqitch was meant to be used more like what you describe here:

> Do we want to push more of this down to data solutions - you can definitely use subdirectories and symlinks, for example, for shared migration piece. Alternately copy-paste, if each project is really a separate fork at the time-of-creation, could make sense in this case.

So I do agree that we should be wary of using this role to orchestrate migrations.  We should let sqitch do that.

Though as noted in your edit we should let it be possible to run multiple migrations e.g. in the case of discover.ona.io we may have migrations for more than one projects living in the same ansible inventory.